lib/pages/bucket_page/bucket_page.dart

```diff
@@ -70,7 +77,38 @@ class BucketPage extends StatelessWidget {
                         size: 22,
                       ),
                       title: Text(objectName),
-                      onPressed: () {},
+                      onPressed: () async {
+                        var downloadDir =
+                            await path_provider.getDownloadsDirectory();
+                        String? selectedDirectory = await FilePicker.platform
+                            .getDirectoryPath(
+                                initialDirectory: downloadDir?.path ?? '/');
+
+                        if (selectedDirectory != null) {
+                          Minio minio = Minio(
+                              endPoint: connection.endpoint,
+                              accessKey: connection.accessKey,
+                              secretKey: connection.secretKey);
+
+                          await minio.fGetObject(
+                              bucket.name,
+                              object.key!,
+                              path_lib.join(selectedDirectory!,
+                                  path_lib.basename(object.key!)));
+
+                          displayInfoBar(context, builder: (context, close) {
+                            return InfoBar(
+                              title: const Text('File downloaded'),
+                              content: Text(object.key!),
+                              action: IconButton(
+                                icon: const Icon(FluentIcons.clear),
+                                onPressed: close,
+                              ),
+                              severity: InfoBarSeverity.success,
```

Comment:
![image](https://user-images.githubusercontent.com/38831366/232550560-cfe61b0b-65b1-4930-96ca-e4b6506a81ea.png)

idk how fix it